Examples/src/TrkGeoToGdmlDumpSvc.cpp

```diff
@@ -0,0 +1,88 @@
+#include "TrkGeoToGdmlDumpSvc.h"
+
+#include "TGeoManager.h"
+
+
+#include "GaudiKernel/ISvcLocator.h"
+#include "GaudiKernel/ServiceHandle.h"
+
+#include "ACTS/Detector/TrackingGeometry.hpp"
+#include "ACTS/Detector/TrackingVolume.hpp"
+#include "ACTS/Tools/CylinderVolumeHelper.hpp"
+#include "ACTS/Tools/CylinderVolumeBuilder.hpp"
+#include "ACTS/Tools/LayerArrayCreator.hpp"
+#include "ACTS/Layers/Layer.hpp"
+#include "ACTS/Tools/TrackingVolumeArrayCreator.hpp"
+#include "ACTS/Examples/BuildGenericDetector.hpp"
+
+DECLARE_SERVICE_FACTORY(TrkGeoToGdmlDumpSvc)
+
+TrkGeoToGdmlDumpSvc::TrkGeoToGdmlDumpSvc(const std::string& aName, ISvcLocator* aSL) : base_class(aName, aSL) {
+  declareProperty("gdml", m_gdmlFileName = "TrkDetector.gdml");
+}
+
+void TrkGeoToGdmlDumpSvc::dumpTrackingVolume(const Acts::TrackingVolume* vol, TGeoVolume* top) {
+  info() << "Dumping TrackingVolume" << vol->volumeName() << endmsg;
+  auto volVec = vol->confinedVolumes();
+  if (nullptr != volVec) {
+    auto objVec = volVec->arrayObjects();
+    for (auto o : objVec) {
+      dumpTrackingVolume(o.get(), top);
+    }
+  }
+  auto layArr = vol->confinedLayers();
+  if (nullptr != layArr) {
+    auto layVec = layArr->arrayObjects();
+    info() << "\t containing " << layVec.size() << " Layers" << endmsg;
+    for (auto l : layVec) {
+      dumpTrackingLayer(l.get(), top);
+    }
+  }
+}
+
+void TrkGeoToGdmlDumpSvc::dumpTrackingLayer(const Acts::Layer* layer, TGeoVolume* top) {
+  auto surfArr = layer->surfaceArray();
+
+  // temporary TGeoVolume that represents one module
+  /// @todo: get dimensions from surface
+  int module_counter = 0;
+  TGeoMaterial* matAl = new TGeoMaterial("Aluminium", 26, 13, 3);
```

Comment:
Aluminium is used in the ROOT TGeo examples ;-) I'll check if there is an easy way to convert the material from Acts.